### PR TITLE
Jxp chapter13

### PR DESCRIPTION
**Pull request by JPeace2**

### CODE REVIEW

File: jxp/Animal/lib/Horse.pm [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+=cut
+
+sub sound { 'neigh' }
+
+=head2 function2
+
+=cut
+
+sub function2 {
+}
```
Comment (dickyince) [inline jxp/Animal/lib/Horse.pm:50]: Empty function is empty

File: jxp/Animal/lib/Cow.pm [hunk 2]
```diff
@@ -3,6 +3,7 @@ package Cow;
 use 5.006;
```
Comment (dickyince) [inline jxp/Animal/lib/Cow.pm:3]: is this version of perl compatible with use parent?

File: jxp/Animal/lib/Person.pm [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+=head2 function2
+
+=cut
+
+sub speak {
+  my $class = shift;
+  my $message = shift;
+  
+  if ( defined $message ){
```
Comment (dickyince) [inline jxp/Animal/lib/Person.pm:53]: is this code repeated in Animal?

Could this logic have been moved to LivingCreature?

File: jxp/Animal/lib/Person.pm [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    print $message;
```
Comment (dickyince) [inline jxp/Animal/lib/Person.pm:54]: Could have been a bit more verbose, something like .... says $message .... maybe

File: jxp/scripts/ch13ex3.pl [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+use strict;
+use warnings;
+use Person;
+use v5.10;
+
+say "Without parameter";
+Person->speak();
+
+say "With parameter";
+Person->speak("Hello World!");
```
Comment (dickyince) [inline jxp/scripts/ch13ex3.pl:11]: Would have been good to include an animal trying to say hello to prove that the error is thrown correctly

File: jxp/Animal/lib/Cow.pm (hunk 2 above)
Comment (alexdeas) [inline jxp/Animal/lib/Cow.pm:3]: Apparently it is.

File: jxp/ch13ex2.pl [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+use v5.10;
+
+say "Please enter a list of Animals";
+my $input = <STDIN>;
+
+my @animals = split " ",$input;
+
+foreach my $animal ( @animals ){
+  $animal->speak();
+}
```
Comment (alexdeas) [inline jxp/ch13ex2.pl:17]: Not too bad, although this might break if someone includes an animal that doesn't exist.

File: jxp/ch13ex2.pl [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+use Cow;
+use Horse;
+use Sheep;
+use Mouse;
+use v5.10;
+
+say "Please enter a list of Animals";
+my $input = <STDIN>;
+
+my @animals = split " ",$input;
```
Comment (utopia-2) [inline jxp/ch13ex2.pl:13]: Reading in `STDIN` into the one string to then split it up seems inefficient. You could build your `@animals` array like:

``` perl
my @animals = do {
  local $/ = ' ';
  <STDIN>
};
```

File: jxp/ch13ex2.pl (hunk 6 above)
Comment (JPeace2) [inline jxp/ch13ex2.pl:17]: Yes, unfortunately I couldn't find a good way to do this. I tried adding ...or print ... some error message, but that didn't work and eval didn't seem to return any error message.

**Pull request merged**